always use linkExpirationFormatter in emails when timePeriodUnit.* are defined

makayabou opened the pull request:
I just noticed that, although ` linkExpirationFormatter.timePeriodUnit` had been defined for some languages in email templates, they were not all correctedd to call directly `linkExpirationFormatter {4}` instead of calling `linkExpiration {1}` and adding **"minutes"**.
Also ,  I corrected permissions on some of those email templates, that had 755 instead of 644.

thomasdarimont committed:
KEYCLOAK-9551 KEYCLOAK-16159 Make refresh_token generation for client_credentials optional. Support for revocation of access tokens.

Co-authored-by: mposolda <mposolda@gmail.com>

zak905 committed:
KEYCLOAK-953: add allowing user to delete his own account feature

psztoch committed:
KEYCLOAK-14492 More polish translation and fixed language name translation

Language name should not be translated!
You search your language name in native.

New messages for lack fields from OpenID connect scopes.

miquelsi committed:
KEYCLOAK-14109. Added profiles for Spring 2.3 version.
KEYCLOAK-14737. Updated Jetty version to 9.4.29, as required per Spring 2.3.

douglaspalmer committed:
[KEYCLOAK-14352] JavaScript injection vulnerability of Realm registration REST API

vmuzikar committed:
KEYCLOAK-14306 OIDC redirect_uri allows dangerous schemes resulting in potential XSS

(cherry picked from commit e86bec81744707f270230b5da40e02a7aba17830)

Conflicts:
    testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientRegistrationTest.java
    testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/ClientTest.java
    services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java

mabartos committed:
KEYCLOAK-15841 Upgrade rest of the minor forms to PF4

Commit:
[KEYCLOAK-15692] Upgrade to Wildfly "21.0.1.Final"

Base fixes:
* [KEYCLOAK-15780]      Upgrade Keycloak to Wildfly 21.0.0.Beta1 / Wildfly Core 13.0.0.Beta6
* [KEYCLOAK-16031]      Upgrade Keycloak to Wildfly 21.0.0.Final / Wildfly Core 13.0.1.Final
* [KEYCLOAK-16442]      Upgrade Keycloak to Wildfly 21.0.1.Final / Wildfly Core 13.0.3.Final

Other (dependent) fixes:
* [KEYCLOAK-15408]      Deprecate former Wildfly and Wildfly Core versions in Arquillian's
                        testsuite pom.xml file as part of the upgrade script
* [KEYCLOAK-15442]      Update the version of 'jboss-parent' as part of the Wildfly upgrade
                        script if necessary
* [KEYCLOAK-15474]      Add --verbose and --force options to the Wildfly upgrade automated script
* [KEYCLOAK-15649]      Update "urn:jboss:domain:infinispan:10.0" version as part of the Wildfly
                        upgrade automated script
* [KEYCLOAK-15652]      Wildfly upgrade automated script - Align Python artifact version
                        comparsion algorithm with the Maven / Java one

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

stianst committed:
KEYCLOAK-15012 Fix issue with folder theme provider

vmuzikar committed:
KEYCLOAK-15295 User can manage resources with just "view-profile" role using new Account Console

(cherry picked from commit 1b063825755d9f5aa13e612757e8ef7299430761)

Commit:
[KEYCLOAK-15571] Fix IdentityProviderFactory.createConfig signature

lhanusov committed:
KEYCLOAK-14231 - validate supported locales

mabartos committed:
KEYCLOAK-14915 Upgrade registration screen to PF4

thomas-riccardi committed:
Finish renaming 'application role' to 'client role' in help texts

martin-kanis committed:
KEYCLOAK-15889 Streamification of ProtocolMappers

Pepo48 committed:
KEYCLOAK-15386 Some icons are not displayed properly

* pficon.woff/woff2 - updated PatternFly font resource
* pficon.css - css that contains @font-face rule
Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

Pepo48 committed:
KEYCLOAK-14137 Upgrade Authentication selector screen

* edited related css and ftl theme resources
* added IE11 compatibility support
* fixed affected tests

Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

Commit:
[KEYCLOAK-16055] Update DefaultKeyManager kid is null logging

Got this "kid is null, can't find public key" without a hint to which realm it's belonging. Not sure if the realm name is dropped because it's null(?), but at least the log message is now explicit. Dropping kid because the text says it's null. Haven't tested whether this breaks tests etc.

tnorimat committed:
KEYCLOAK-14201 Client Policy - Executor : Enforce Proof Key for Code Exchange (PKCE)

tnorimat committed:
KEYCLOAK-14199 Client Policy - Executor : Enforce more secure client authentication method when client registration

ibacher committed:
KEYCLOAK-15287 Ability to add custom claims to the AccessTokenResponse

Pepo48 committed:
KEYCLOAK-14138 Upgrade OTP login screen

* edited related css and ftl theme resources
* added tile component
* fixed IE11 compatibility
* fixed affected tests

Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

sguilhen committed:
[KEYCLOAK-15200] Complement methods for accessing users with Stream variants

miquelsi committed:
KEYCLOAK-13639. Added metrics and custom healthcheck endpoints, both enabled via 'metrics.enabled' config parameter.

rmartinc committed:
[KEYCLOAK-16139] Serialize the calls for groups tab on admin console

lscorcia committed:
KEYCLOAK-16065 Replace last UrlConnection uses with HttpClientProvider

martin-kanis committed:
KEYCLOAK-15898 Streamification of Keymanager

pedroigor committed:
[KEYCLOAK-14468] - Scope permission sometimes not removed when removing scopes

micedre committed:
KEYCLOAK-16112 don't remove username attribute

hmlnarik committed:
KEYCLOAK-16115 Remove need for MapStorage.keySet() and values()

makayabou committed:
always use linkExpirationFormatter in emails when timePeriodUnit.* are defined

makayabou commented:
Sorry I'm new to PR.
I saw Keycloak CI failing, is it a problem? 
Is it because i didn't named the branch I worked on?
How can I push changes to PR to have CI checks success ?

Thanks

mabartos commented:
@makayabou Thanks for your PR! However, you should create an JIRA issue. Please, see [here](https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md#create-an-issue-in-keycloak-jira).

pedroigor committed:
[KEYCLOAK-16116] - PersistedConfigSource for resolving build time properties

pedroigor committed:
[KEYCLOAK-11698] - Change context path of Keycloak to / for Keycloak.X

hmlnarik committed:
KEYCLOAK-16090 Work around LDAPUserLoginTest false failures

martin-kanis committed:
KEYCLOAK-16145 Add adapter-tests to GitHub Actions

mabartos committed:
KEYCLOAK-15242 Error in application logs when refresh token is expired.

stianst committed:
KEYCLOAK-16144 Added unit tests to GA

martin-kanis committed:
KEYCLOAK-16166 Remove adapter-tests from GitHub Actions

stianst committed:
KEYCLOAK-16167 Enable Account REST API by default

martin-kanis committed:
KEYCLOAK-16147 KEYCLOAK-16148 Add crossdc tests to GitHub Actions

pedroigor committed:
Minor updates to config based on design document changes

hmlnarik committed:
KEYCLOAK-16156 Optimize maven repo handling

martin-kanis committed:
KEYCLOAK-16155 Add cluster tests to GitHub Actions

hmlnarik committed:
KEYCLOAK-16189 Publish surefire reports

miquelsi committed:
KEYCLOAK-13644. Added healthcheck endpoints tests.

pedroigor committed:
[KEYCLOAK-16126] - Windows support

tnorimat committed:
KEYCLOAK-14198 Client Policy - Condition : Client - Client IP

hmlnarik committed:
KEYCLOAK-16217 Fix compilation errors

hmlnarik committed:
KEYCLOAK-16157 Fix Unexpected I/O error message

edwint88 committed:
[KEYCLOAK-16215] Typo in EventConfigTest

tnorimat committed:
KEYCLOAK-14206 Client Policy - Executor : Enforce more secure state and nonce treatment for preventing CSRF

mposolda committed:
KEYCLOAK-16244 RealmRealmLocalizationResourceTest fails on auth-server-quarkus

tnorimat committed:
KEYCLOAK-14196 Client Policy - Condition : Client - Client Scope

sguilhen committed:
[KEYCLOAK-16232] Replace usages of deprecated collection-based methods with the respective stream variants

tnorimat committed:
KEYCLOAK-14207 Client Policy - Executor : Enforce more secure client signature algorithm when client registration

pedroigor committed:
[KEYCLOAK-16245] - Update Quarkus 1.10.0.CR1

hmlnarik committed:
KEYCLOAK-16297 Fix HttpClient stale connections

pedroigor committed:
[KEYCLOAK-13639] - Improvements to metrics and health status

hmlnarik committed:
KEYCLOAK-16158 Fix error reporting

pedroigor committed:
[KEYCLOAK-11330] - Fixing client and realm tests

hmlnarik committed:
KEYCLOAK-16323 Speedup GHA

mabartos committed:
KEYCLOAK-16143 Login form expected, but registraion form is displayed

mhajas committed:
KEYCLOAK-16074 Fix check3pCookiesSupported message callback

thomasdarimont committed:
KEYCLOAK-16143 Honor AuthenticationProcessor.forwardedErrorMessage when rendering registration form

tnorimat committed:
KEYCLOAK-14193 Client Policy - Condition : Client - Client Access Type

lscorcia committed:
KEYCLOAK-16325 Introduce SAML2NameIDBuilder for easier creation of NameIDType elements

lscorcia committed:
KEYCLOAK-16325 Forgot the NameQualifier property in SAML2NameIDBuilder

sguilhen committed:
[KEYCLOAK-16341] Make the new stream-based methods in server-spi user interfaces default instead of the collection-based versions.
 - this ensures that providing implementation for the collection-based methods is enough, which preserves
   backwards compatibility with older custom implementations.
 - alternative interfaces now allow new implementations to focus on the stream variants of the query methods.

ssilvert committed:
KEYCLOAK-16354: Update serialize-javascript dependency

ssilvert committed:
KEYCLOAK-14547: Make New Account Console the default.

nkkumawat committed:
KEYCLOAK-16381: error text moved to constants file

tnorimat committed:
KEYCLOAK-14197 Client Policy - Condition : Client - Client Host

tnorimat committed:
KEYCLOAK-16392 Client Policy - Condition : NPE without any initial configuration

vmuzikar committed:
KEYCLOAK-14547: Make New Account Console the default.

Commit:
KEYCLOAK-11699 add support for 127.0.0.1 for native app

hmlnarik committed:
KEYCLOAK-16405 Tests for storage logical layer

lscorcia committed:
KEYCLOAK-16429 Pass default boolean values as strings, as expected by the UI

hmlnarik committed:
KEYCLOAK-16444 Initialize JAXP components consistently

martin-kanis committed:
KEYCLOAK-14556 Authentication session map store

sguilhen committed:
[KEYCLOAK-16232] Streamify the UserCredentialStore and UserCredentialManager interfaces

growingspaghetti committed:
KEYCLOAK-16474 typo in javadoc sproxy_set_header -> proxy_set_header

vanrar68 committed:
KEYCLOAK-15634: Prevent federated user removal with new account console

mabartos committed:
KEYCLOAK-15264 Import realm using directory provider twice with IGNORE_EXISTING will cause NPE for clientId

vanrar68 committed:
KEYCLOAK 15634: Add ldap test

hmlnarik committed:
KEYCLOAK-16489 Add ability to run model tests with LDAP

vramik committed:
KEYCLOAK-16502 Migration of DELETE_ACCOUNT role

mabartos committed:
KEYCLOAK-15066 Internal Server error when calling random idp endpoint

sguilhen committed:
[KEYCLOAK-16508] Complement methods for accessing user sessions with Stream variants

lscorcia committed:
KEYCLOAK-16519 Fix typo in regex

The regex has a typo that prevents correct splitting of parameter values containing multiple OIDs.

tnorimat committed:
KEYCLOAK-14191 Client Policy - Condition : Author of a client - User Group

vramik committed:
KEYCLOAK-15390 fix ClientMappersOIDCTest

vramik committed:
KEYCLOAK-16543 fix compilation failure on keycloak-services

tnorimat committed:
KEYCLOAK-14192 Client Policy - Condition : Author of a client - User Role

Commit:
[KEYCLOAK-16551] Remove references to Gatekeeper in the README file

martin-kanis committed:
KEYCLOAK-11417 Internal server error on front channel logout with expired session

vmuzikar committed:
KEYCLOAK-16533 Improve FluentTestsHelper to better support QS testing

hmlnarik committed:
KEYCLOAK-16567 Optimize StackUtil class

mabartos committed:
KEYCLOAK-16583 Ignore tests which directly use WebAuthn Chrome testing feature

mhajas committed:
KEYCLOAK-15847 Add MapUserProvider

tnorimat committed:
KEYCLOAK-14203 Client Policy - Executor : Enforce HTTPS URIs

fabrc committed:
Correct label is 'Dependent Permissions' (#7672)

The label refers to the list 'permissions' dependent on the 'policy'.

stianst committed:
Fix for distribution

keycloak-bot committed:
Set version to 13.0.0-SNAPSHOT

hmlnarik committed:
KEYCLOAK-16648 Fix migration to 12
